libres/lib/analysis/update.cpp

```diff
@@ -0,0 +1,680 @@
+                             enkf_fs_type *fs, const char *key, int report_step,
+                             const active_list_type *active_list) {
+    const enkf_config_node_type *config_node =
+        ensemble_config_get_node(ensemble_config, key);
+    /*
+     This is very awkward; the problem is that for the GEN_DATA
+     type the config object does not really own the size. Instead
+     the size is pushed (on load time) from gen_data instances to
+     the gen_data_config instance. Therefor we have to assert
+     that at least one gen_data instance has been loaded (and
+     consequently updated the gen_data_config instance) before we
+     query for the size.
+  */
+    {
+        if (enkf_config_node_get_impl_type(config_node) == GEN_DATA) {
+            enkf_node_type *node = enkf_node_alloc(config_node);
+            node_id_type node_id = {.report_step = report_step, .iens = 0};
+
+            enkf_node_load(node, fs, node_id);
+            enkf_node_free(node);
+        }
+    }
+
+    {
+        active_mode_type active_mode = active_list_get_mode(active_list);
+        int active_size;
+        if (active_mode == INACTIVE)
+            active_size = 0;
+        else if (active_mode == ALL_ACTIVE)
+            active_size =
+                enkf_config_node_get_data_size(config_node, report_step);
+        else if (active_mode == PARTLY_ACTIVE)
+            active_size = active_list_get_active_size(active_list, -1);
+        else {
+            util_abort("%s: internal error .. \n", __func__);
+            active_size = -1; /* Compiler shut up */
+        }
+        return active_size;
+    }
+}
+
+static void serialize_node(enkf_fs_type *fs,
+                           const enkf_config_node_type *config_node, int iens,
+                           int report_step, int row_offset, int column,
+                           const active_list_type *active_list,
+                           matrix_type *A) {
+
+    enkf_node_type *node = enkf_node_alloc(config_node);
+    node_id_type node_id = {.report_step = report_step, .iens = iens};
+    enkf_node_serialize(node, fs, node_id, active_list, A, row_offset, column);
+    enkf_node_free(node);
+}
+
+static void *serialize_nodes_mt(void *arg) {
+    serialize_info_type *info = (serialize_info_type *)arg;
+    const auto *node_info = info->node_info;
+    const enkf_config_node_type *config_node =
+        ensemble_config_get_node(info->ensemble_config, node_info->key);
+    for (int iens = info->iens1; iens < info->iens2; iens++) {
+        int column = int_vector_iget(info->iens_active_index, iens);
+        if (column >= 0) {
+            serialize_node(info->src_fs, config_node, iens, info->report_step,
+                           node_info->row_offset, column,
+                           node_info->active_list, info->A);
+        }
+    }
+    return NULL;
+}
+
+static void enkf_main_serialize_node(const char *node_key,
```

Comment:
Remove `enkf_main` from name?

Reply:
Its quite common in c-style programming (no cpp) to prefix function names with a pseudo-namespace (this is just to avoid name collisions as every function resides in the same namespace in c), in this it used to reside in the "enkf_main" namespace but would now more properly reside in the `analysis` namespace. Therefore I think it should be replaced by `analysis`/`analysis_update` or the like.